internal/batches/executor/executor.go

```diff
@@ -176,6 +176,8 @@ func (x *executor) do(ctx context.Context, task *Task, ui TaskExecutionUI) (err
 		reportProgress: func(currentlyExecuting string) {
 			ui.TaskCurrentlyExecuting(task, currentlyExecuting)
 		},
+		newUiStdoutWriter: ui.StepStdoutWriter,
+		newUiStderrWriter: ui.StepStderrWriter,
```

Comment:
The idea is that in the near future we don't pass 3+ things to `runSteps`, but instead just a single thing that implements something like `StepsExecutionUI`, which has methods for different events: `DownloadingArchive()`, `StartingStep()`, `FinishedStep()`, etc.

But that seems a bit out of scope of this PR and we still don't know 100% what exactly we need from the JSON output in the future, since we might be able to get around some logging once we have more information about which Task gets executed on the server-side.